### DISCUSSION TITLE
Rotate Table header text 90 degrees

### DISCUSSION THREAD

**Discussion by Maurice-van-Beers** [EvotecIT/PSWriteHTML#323] [General]
How can i rotate the text of an Table Column header 90 degrees

**Comment by PrzemyslawKlys** [comment]
You probably need to do 3 things:
1. Insert custom CSS in HTML using `Add-HTMLStyle` that is based on this: https://stackoverflow.com/questions/47261100/how-to-rotate-text-90-degrees-inline
2. Then in your PowerShell data you would need to replace your property names with <div class="thatyoudefined">yourproperty<div>
3. Finally use `InvokeHTMLTags` for `New-HTMLTable`

**Reply by Maurice-van-Beers** [reply]
I've tried this, but i'm not doing it right, could you please show me an example.

```powershell
$Process = Get-Process | Select-Object -First 20
$ProcessesLimited = $Process | Select-Object -Property Id, Name, PriorityClass, ProductVersion, Company, Path, Description
New-HTML -TitleText 'My title' -Online -FilePath $PSScriptRoot\Example39_01.html {
    Add-HTMLStyle -Content "#rotate-text {width: 20px; transform: rotate(90deg);}" 
    New-HTMLTable -InvokeHTMLTags {"div id=`"rotate-text`""} -DataTable $ProcessesLimited
} -ShowHTML
```

**Reply by PrzemyslawKlys** [reply]
It doesn't work that way. InvokeHTMLTags is a switch. It turns on/off HTML invoking, but to get this to work you need to rewrite every single property in the ProcessesLimited.

```powershell

$Process = Get-Process | Select-Object -First 3
$ProcessesLimited = $Process | Select-Object -Property Id, Name, PriorityClass, ProductVersion, Company, Path, Description
$Properties = $ProcessesLimited[0].PSObject.Properties.Name
[Array] $ModifiedObjects = foreach ($P in $ProcessesLimited) {
    $NewObject = [ordered]@{}
    foreach ($Property in $Properties) {
        $NewObject["<div class = `"rotate-text`">$Property</div>"] = $P.$Property
    }
    [PSCUstomObject] $NewObject
}

New-HTML -TitleText 'My title' -Online -FilePath $PSScriptRoot\Example39_01.html {
    Add-HTMLStyle -Content ".rotate-text { width: 20px; transform: rotate(90deg); }" -SkipTags
    New-HTMLTable -InvokeHTMLTags -DataTable $ModifiedObjects
} -ShowHTML
```

![image](https://user-images.githubusercontent.com/15063294/173844438-5f49ca3d-4fec-44fa-8fc7-3b59b43a9549.png)
